doc/nvim-tree-lua.txt

```diff
@@ -1,90 +1,125 @@
 *nvim-tree.lua* A file explorer written in lua
 
-Minimum version of neovim: 0.6.0
-
 Author: Yazdani Kiyan <yazdani.kiyan@protonmail.com>
 
 ==============================================================================
-INTRODUCTION                                         *nvim-tree-introduction*
+CONTENTS                                                          *nvim-tree
 
-This file explorer requires `neovim >= 0.6.0`
+  1. Introduction                             nvim-tree-introduction
+  2. Quickstart                               nvim-tree-quickstart
+  3. Commands                                 nvim-tree-commands
+  4. Setup/Configuration                      nvim-tree-setup
+  5. Mappings                                 nvim-tree-mappings
+  6. Highlight Groups                         nvim-tree-highlight
+  7. Events                                   nvim-tree-events
 
 ==============================================================================
-QUICK START                                          *nvim-tree-quickstart*
+ 1. INTRODUCTION                                     nvim-tree-introduction
 
-open the tree with :NvimTreeToggle
->
-  nnoremap <C-n> :NvimTreeToggle<CR>
-  lua require 'nvim-tree'.toggle()
-<
+<add shortish description here>
+
+Features:
+    File icons with vim-devicons.
+
+    Uses other type of icons so a good font support is recommended.
+    If the tree renders weird glyphs, install the correct fonts.
+
+    Syntax highlighting uses g:terminal_color_ from colorschemes, fallbacks to
+    ugly colors otherwise.
+
+    Git integration tells when a file is:
+    - ✗  unstaged or folder is dirty
+    - ✓  staged
+    - ★  new file
+    - ✓ ✗ partially staged
+    - ✓ ★ new file staged
+    - ✓ ★ ✗ new file staged and has unstaged modifications
+    - ═  merging
+    - ➜  renamed
+
+    Mouse support defined in |KeyBindings|
+
+Requirements:
+    This file explorer requires `neovim >= 0.7.0`
+
+==============================================================================
+ 2. QUICK START                                         *nvim-tree-quickstart*
+
+Configure: TODO: REPHRASE
+
+    require('nvim-tree').setup()
+    vim.keymap.add('n', <C-n>', require('nvim-tree').toggle)
+
+open the tree with :NvimTreeToggle or <C-n>
 
 ==============================================================================
-COMMANDS                                             *nvim-tree-commands*
+ 3. COMMANDS                                             *nvim-tree-commands*
```

Comment:
Anything that's an M.something is API: [nvim-tree.lua/lua/nvim-tree.lua](https://github.com/kyazdani42/nvim-tree.lua/blob/1caca6285427ebd2b4f0eb10f4d1ae3956ff09c0/lua/nvim-tree.lua#L259)

It would be great to mention them here, as it allows us to also document their arguments.

We could have a "public API" section, but I think this will do for now.